src/App.tsx

```diff
@@ -8,53 +8,103 @@
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUser()
+      .then(setUsers)
+      .catch(() => setIsError(true));
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    if (activeUser) {
+      setIsLoading(true);
+      getPosts({ id: activeUser.id })
+        .then(setPosts)
+        .catch(() => setIsError(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [activeUser]);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const isPostsList = activeUser && !isError && posts;
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  activeUser={activeUser}
+                  setActiveUser={user => {
+                    setActiveUser(user);
+                    setActivePost(null);
+                  }}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!activeUser && !isError && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                <Loader isLoading={isLoading} />
+                {isError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
+
+                {!isLoading && !posts.length && !isError && activeUser && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
+                {!isLoading && posts.length > 0 && (
+                  <PostsList
+                    isPostsList={isPostsList}
```

Comment:
The `isPostsList` prop is being passed to the `PostsList` component, but it's not clear if this prop is used within `PostsList`. Ensure that `PostsList` is expecting this prop and using it appropriately. If not, consider removing it or updating `PostsList` to handle this prop.